extensions/resteasy-reactive/quarkus-resteasy-reactive-links/deployment/src/main/java/io/quarkus/resteasy/reactive/links/deployment/LinksProcessor.java

```diff
@@ -117,19 +121,49 @@ private RuntimeValue<GetterAccessorsContainer> implementPathParameterValueGetter
             for (LinkInfo linkInfo : linkInfos) {
                 String entityType = linkInfo.getEntityType();
                 for (String parameterName : linkInfo.getPathParameters()) {
+                    DotName className = DotName.createSimple(entityType);
+                    FieldInfoSupplier byParamName = new FieldInfoSupplier(c -> c.field(parameterName), className, index);
+
                     // We implement a getter inside a class that has the required field.
                     // We later map that getter's accessor with an entity type.
                     // If a field is inside a parent class, the getter accessor will be mapped to each subclass which
                     // has REST links that need access to that field.
-                    FieldInfo fieldInfo = getFieldInfo(index, DotName.createSimple(entityType), parameterName);
+
+                    FieldInfo fieldInfo = byParamName.get();
+                    if ((fieldInfo == null) && parameterName.equals("id")) {
+                        // this is a special case where we want to go through the fields of the class
+                        // and see if any is annotated with any sort of @Id annotation
+                        // N.B. as this module does not depend on any other module that could supply this @Id annotation
+                        // (like Panache), we need this general lookup
+                        FieldInfoSupplier byIdAnnotation = new FieldInfoSupplier(
+                                c -> {
+                                    for (FieldInfo field : c.fields()) {
+                                        List<AnnotationInstance> annotationInstances = field.annotations();
+                                        for (AnnotationInstance annotationInstance : annotationInstances) {
+                                            if (annotationInstance.name().toString().toLowerCase(Locale.ROOT)
```

Comment:
I think that this logic is too wide, maybe we should be more strict by using the dot name of `javax.persistence.Id`.

Reply:
I thought about it and I really don't know. If we did that, it would pretty much mean that this can only be used with Hibernate REST Data Panache.

Reply:
I guess we can start by being strict

Reply:
Done